### ISSUE TITLE
[HOLD for payment 2023-08-03] [$1000] Emoji picker's position changes on decreasing the screen size

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#17603]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open the app
2. Go to any chat > Open Emoji Picker
3.  Decrease the screen size like shown in video
 4.   Notice that emoji picker's position changes

## Expected Result :
The emoji picker jumps to the LHN instead of behaving the correct way (as in the second video).

## Actual Result : 
 Emoji picker's position changes

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.1
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

Current:

https://user-images.githubusercontent.com/43996225/232853390-155ddcda-1f55-4bf2-b545-974fe24af70d.mov

Expected:

https://user-images.githubusercontent.com/43996225/232853411-8ec3b9a6-dfc0-40d8-9005-6612ce1fe320.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @daraksha-dk 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1681758191524309

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~019f4f66f49e6e10ed</li>
        <li>Upwork Job ID: 1676073590122438656</li>
        <li>Last Price Increase: 2023-07-04</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @sophiepintoraetz (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by sophiepintoraetz** [comment]
This is not an issue but expected behaviour due to the screen resizing.

**Comment by kadiealexander** [comment]
Hey Sophie! The first video in the image is the one that shows the bug. The emoji picker jumps to the LHN instead of behaving the correct way (as in the second video). This was also reported [here](https://github.com/Expensify/App/issues/21982) but I'm reopening this one instead to ensure the original bug reporter is credited.

**Comment by sophiepintoraetz** [comment]
Thanks, @kadiealexander - that description helped!
>The emoji picker jumps to the LHN instead of behaving the correct way (as in the second video).

**Comment by alitoshmatov** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

### What is the root cause of that problem?
This is caused by this line where we are checking if device is touchscreen and medium size, if so we are removing emoji picker button
https://github.com/Expensify/App/blob/ecfcd8325d73851499f00399b14b8ae8e7f55f90/src/pages/home/report/ReportActionCompose.js#L1123

Thus when this button disappears emoji picker popover will lose it is target position
### What changes do you think we should make in order to solve the problem?
I am not sure about this line: 
https://github.com/Expensify/App/blob/ecfcd8325d73851499f00399b14b8ae8e7f55f90/src/pages/home/report/ReportActionCompose.js#L1123
It is not logical to me, for example in mid sized tablets we do not show emoji picker button, how are they going to use it then?
This is image of ipad mini in chrome responsive mode
<img width="810" alt="Screenshot 2023-07-04 at 11 53 06 AM" src="https://github.com/Expensify/App/assets/59907218/d94e099e-7a68-4ecb-8102-08892a68eeae">
I think we should remove this condition and want to show emoji picker button always.

### What alternative solutions did you explore? (Optional)

**Comment by sophiepintoraetz** [comment]
Just waiting on @allroundexperts to review!

**Comment by alitoshmatov** [comment]
@allroundexperts You have to simulate in responsive mode of chrome, this issue is reproducible only in medium sized screen and touchscreen enabled

**Comment by alitoshmatov** [comment]
@allroundexperts 

https://github.com/Expensify/App/assets/59907218/16bc4a83-fb06-4bcc-be32-a12dfec9c823

**Comment by alitoshmatov** [comment]
@allroundexperts What's your opinion on this, can we confirm on this behavior. 

> It is not logical to me, for example in mid sized tablets we do not show emoji picker button, how are they going to use it then?
This is image of ipad mini in chrome responsive mode
<img width="810" alt="Screenshot 2023-07-04 at 11 53 06 AM" src="https://github.com/Expensify/App/assets/59907218/d94e099e-7a68-4ecb-8102-08892a68eeae">

**Comment by alitoshmatov** [comment]
Sorry missed that 👍

**Comment by Li357** [comment]
I'll hop in and say I prefer @hoangzinh's proposal, and if we want to show the picker on other sizes we should do that in a separate PR. It makes more sense to use what React's lifecycle provides here to hide a related component on unmount. Is there a case where unmount != invisible?

**Comment by alitoshmatov** [comment]
You are right @hoangzinh approach looks better. But it does not solve this issue https://github.com/Expensify/App/issues/21982 . Or any instance where emoji picker menu loses it is anchor element.

Also, current issue is only reproducible only in few devices which has touchscreen capabilities, and it eliminates safari browser.

**Comment by Li357** [comment]
Hmm, what about mobile Safari? We can check `emojiPopoverAnchor` in a useEffect as well right?

**Comment by alitoshmatov** [comment]
We won't be having this issue in mobile safari since window won't be resized, and drawer component will be used to show emojis. 

> Also we can hide emoji picker if anchor element is unmounted. I checked if emojiPopoverAnchor.current will be null if referenced element is unmounted but no it is still pointing to the element, but the element is not present in dom.

Checking `emojiPopoverAnchor` in useEffect will not work because even if original element is unmounted, the element is still present in `emojiPopoverAnchor`. Would be great, if we could check element's status from what we have in `emojiPopoverAnchor`

**Comment by alitoshmatov** [comment]
@hoangzinh That looks promising 👍

**Comment by sophiepintoraetz** [comment]
@allroundexperts - sorry for the delay here! I agree with your suggestion here:
> I prefer to hide the emoji picker menu when the button gets hidden instead of showing the button always.

**Comment by alitoshmatov** [comment]
@hoangzinh Did you test you solution to see whether it works?

@allroundexperts I think you should wait @hoangzinh confirmation if his proposed solution works on all cases.

**Comment by jliexpensify** [comment]
@allroundexperts I don't think I'm assigned this GH?

**Comment by sophiepintoraetz** [comment]
Still in review.

**Comment by sophiepintoraetz** [comment]
Payouts due (I believe the date for the bonus calculation is correct):

Issue Reporter: $250 @daraksha-dk
Contributor: $500 @hoangzinh r (assigned 13 Jul - PR merged 26 Jul = 14 working days lapsed)
Contributor+: $500 @ @allroundexperts (assigned 13 Jul - PR merged 26 Jul = 14 working days lapsed)

Eligible for 50% #urgency bonus? N (penalty applied)

**Comment by sophiepintoraetz** [comment]
@hoangzinh I appreciate the ask but I do not think that's fair to other individuals who submitted proposals and could have completed this job/other contributors who have been penalised for the same reasons. I am glad that you were thorough but we do make the guidelines clear - I would suggest taking more care in your original proposal!

**Comment by sophiepintoraetz** [comment]
@allroundexperts - sorry - I used a calculator which I thought was accurate, you're right in that it wasn't 14 days but am I correct in saying that there are 10 days between assignment and merging?
![image](https://github.com/Expensify/App/assets/42940078/64746315-d8ba-431b-8e8f-5d646fa1f82d)

**Comment by sophiepintoraetz** [comment]
Got it. Moot point seeing as we're in agreement but I appreciate the accuracy. Reconfirming that the payments will be
Issue Reporter: $250 @daraksha-dk
Contributor: $500 @hoangzinh r (assigned 13 Jul - PR merged 26 Jul = 14 working days lapsed)
Contributor+: $500 @ @allroundexperts (assigned 13 Jul - PR merged 26 Jul = 14 working days lapsed)

**Comment by sophiepintoraetz** [comment]
Offers sent to @daraksha-dk and @hoangzinh

**Comment by sophiepintoraetz** [comment]
Payments done - @allroundexperts - just drop the request here when you're ready!

**Comment by JmillsExpensify** [comment]
Reviewed the above payment details for @allroundexperts. This is approved for payment in NewDot.
